### PR TITLE
feat(cards): #MAG-8 add side-bar for creating cards

### PR DESCRIPTION
**Pull request by flmartineau**
## Describe your changes

- add side-bar for adding different types of magnets

## Checklist tests

- check if magnet creation is working properly for all types

## Issue ticket number and link

https://jira.support-ent.fr/browse/MAG-8

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/controllers/board-view.controller.ts [hunk 1]
```diff
@@ -154,11 +220,73 @@ class Controller implements IViewModel {
...
+            this.cardForm.resourceUrl = this.$sce.trustAsResourceUrl(
+                this.videoUrl.split("src=\"")[1].split('"')[0]);
+        }
+
+        this.displayCardLightbox = true;
+        safeApply(this.$scope);
+    }
+
+    onLinkSubmit = (form: {url: "", title: ""}) : void => {
+        this.cardForm.resourceUrl = form.url;
```
Comment (sjollois) [inline src/main/resources/public/ts/controllers/board-view.controller.ts:265]: Attention à la vérification qu'il n'y ait pas d'injection de code XSS

**Review by sjollois — CHANGES_REQUESTED** [review]

File: src/main/resources/public/sass/global/components/_icons.scss [hunk 2]
```diff
@@ -115,6 +115,11 @@ i {
     content: '\F387';
   }
 
+  &.magneto-settings::before {
+    font-family: 'material-icons';
```
Comment (Sofianebernoussi) [inline src/main/resources/public/sass/global/components/_icons.scss:119]: font family a changé 'magneto-mdi'

File: src/main/resources/public/template/board.html [hunk 3]
```diff
@@ -50,6 +53,34 @@ <h1 ng-click="vm.goToBoards()" class="cell paddingRight">
...
+            on-submit="vm.onFileSelected">
+        </media-library-lightbox>
+
+        <embedder file-format="'video'"
+                  hidden-share-video-code="false"
+                  ng-model="vm.videoUrl"
+                  ng-change="vm.onVideoSelected()"
+                  ng-if="vm.displayResourceLightbox.video"
+                  show="vm.displayResourceLightbox.video"
+                  style="z-index: 99999;">
```
Comment (Sofianebernoussi) [inline src/main/resources/public/template/board.html:77]: tu veux pas le mettre dans une class plutot?

File: src/main/resources/public/ts/controllers/board-view.controller.ts [hunk 4]
```diff
@@ -154,11 +220,73 @@ class Controller implements IViewModel {
...
     resetCards = (): void => {
         this.filter.page = 0;
         this.cards = [];
     }
 
+    onFileSelected = async (file: any): Promise<void> => {
+        this.displayMediaLibraryLightbox = false;
+        this.cardForm.title = "";
+        this.cardForm.description = "";
+        this.cardForm.caption = "";
```
Comment (Sofianebernoussi) [inline src/main/resources/public/ts/controllers/board-view.controller.ts:238]: pas moyen de faire quelque chose avec le constructeur peut être de cardForm? au lieu de reset les 3

**Review by JinDouang — APPROVED** [review]

**Pull request merged**